test/integration/commands/commandUtils.ts

```diff
@@ -1,4 +1,52 @@
 import { MatrixClient } from "matrix-bot-sdk";
+import { strict as assert } from "assert";
+import * as crypto from "crypto";
+
+/**
+ * Returns a promise that resolves to the first event replying to the event produced by targetEventThunk.
+ * @param client A MatrixClient that is already in the targetRoom that can be started to listen for the event produced by targetEventThunk.
+ * This function assumes that the start() has already been called on the client.
+ * @param targetRoom The room to listen for the reply in.
+ * @param targetEventThunk A function that produces an event ID when called. This event ID is then used to listen for a reply.
+ * @returns The replying event.
+ */
+ export async function getFirstReply(client: MatrixClient, targetRoom: string, targetEventThunk: () => Promise<string>): Promise<any> {
+    let reactionEvents = [];
+    const addEvent = function (roomId, event) {
+        if (roomId !== targetRoom) return;
+        if (event.type !== 'm.room.message') return;
+        reactionEvents.push(event);
+    };
+    let targetCb;
+    try {
+        client.on('room.event', addEvent)
+        const targetEventId = await targetEventThunk();
+        for (let event of reactionEvents) {
+            const in_reply_to = event.content['m.relates_to']?.['m.in_reply_to'];
+            if (in_reply_to.event_id === targetEventId) {
+                return event;
+            }
+        }
+        return await new Promise((resolve, reject) => {
+            targetCb = function(roomId, event) {
+                if (roomId !== targetRoom) return;
+                if (event.type !== 'm.room.message') return;
+                const in_reply_to = event.content['m.relates_to']?.['m.in_reply_to'];
+                if (in_reply_to.event_id === targetEventId) {
+                    resolve(event)
```

Comment:
Shouldn't you unregister here? This would be simpler than the `targetCb` danse, no?